readthedocs/proxito/views/serve.py

```diff
@@ -265,6 +265,10 @@ def serve_path(self, request, path):
         version = unresolved.version
         filename = unresolved.filename
 
+        # Inject the UnresolvedURL into the HttpRequest so we can access from the middleware.
+        # We could resolve it again from the middleware, but we would duplicating DB queries.
+        request.unresolved_url = unresolved
```

Comment:
This should also be in the 404 view.

Reply:
👍🏼 -- I put it there.

I found there could be cases where the `unresolve_path` could raise and exception making the `request.unresolved_url=None` but there would be a `filename` anyways, since it's got from `exc.path`.

I think we could eventually refactor the proxito/unresolver code to support `UnresolvedURL` without _all_ the components. This is out of the scope of this PR, tho and probably also related to https://github.com/readthedocs/readthedocs.org/issues/10456